src/edwards.rs

```diff
@@ -447,32 +446,67 @@ impl ProjectivePoint {
         CompressedEdwardsY(s)
     }
 
-    /// Convert this point to a `CompressedMontgomeryU`.
-    /// Note that this discards the sign.
+    /// Convert this projective point in the Edwards model to its equivalent
+    /// projective point on the Montgomery form of the curve.
     ///
-    /// # Return
-    /// - `None` if `self` is the identity point;
-    /// - `Some(CompressedMontgomeryU)` otherwise.
+    /// Taking the Montgomery curve equation in affine coordinates:
     ///
-    pub fn compress_montgomery(&self) -> Option<CompressedMontgomeryU> {
-        // u = (1 + y) /  (1 - y)
-        // v = sqrt(-486664) * u / x
-        //
-        // since y = Y/Z, x = X/Z,
-        //
-        // u = (1 + Y/Z) / (1 - Y/Z);
-        //   =   (Z + Y) / (Z - Y);
-        //
-        // exceptional points:
-        // y = 1 <=> Y/Z = 1 <=> Z - Y = 0
-        let Z_plus_Y   = &self.Z + &self.Y;
-        let Z_minus_Y  = &self.Z - &self.Y;
-        let u = &Z_plus_Y * &Z_minus_Y.invert();
-
-        if Z_minus_Y.is_zero() == 0u8 {
-            Some(CompressedMontgomeryU(u.to_bytes()))
-        } else {
-            None
+    ///     E_(A,B) = Bv² = u³ + Au² + u   <span style="float: right">(1)</span>
+    ///
+    /// and given its relations to the coordinates of the Edwards model:
+    ///
+    ///     u = (1+y)/(1-y)                <span style="float: right">(2)</span>
+    ///     v = (λu)/(x)
+    ///
+    /// Converting from affine to projective coordinates in the Montgomery
+    /// model, we arrive at:
+    ///
+    ///     u = (Z+Y)/(Z-Y)                <span style="float: right">(3)</span>
+    ///     v = λ * ((Z+Y)/(Z-Y)) * (Z/X)
+    ///
+    /// The transition between affine and projective is given by
+    ///
+    ///      u → U/W                       <span style="float: right">(4)</span>
+    ///      v → V/W
+    ///
+    /// thus the Montgomery curve equation (1) becomes
+    ///
+    ///      E_(A,B) : BV²W = U³ + AU²W + UW² ⊆ 𝗣^2  <span style="float: right">(5)</span>
+    ///
+    /// Here, again, to differentiate from points in the twisted Edwards model, we
+    /// call the point `(x,y)` in affine coordinates `(u,v)` and similarly in projective
+    /// space we use `(U:V:W)`.  However, since (as per Montgomery's original work) the
+    /// v-coordinate is superfluous to the definition of the group law, we merely
```

Comment:
nit: rather than "superfluous", should be something like "scalar multiplication can be computed using only the u-coordinate" since the Montgomery line doesn't inherit the group law from the curve

Reply:
oops, I see that you already did this below, just here the same change is missing

Reply:
Oh, github is showing you an old diff? This was fixed in c08591a7